src/services/rfqm_service.ts

```diff
@@ -469,18 +501,52 @@ export class RfqmService {
             assetFillAmount,
             RFQM_MINIMUM_EXPIRY_DURATION_MS,
         );
+        let makerUri: string = '';
+        let wrappedBestQuote: WrappedSignedNativeOrderMM | null = null;
+        if (bestQuote) {
+            // Get the makerUri
+            const tmpMakerUri = this._quoteRequestor.getMakerUriForSignature(bestQuote.signature);
+            if (tmpMakerUri === undefined) {
+                throw new Error(`makerUri unknown for maker address ${bestQuote.order.maker}`);
+            } else {
+                makerUri = tmpMakerUri;
+            }
+            wrappedBestQuote = { order: bestQuote, makerUri };
+        }
```

Comment:
ahh, ok, logic here will actually change - I think it might be best to wait for the rebase. In the new version, you won't have to attach the makerUri, as it will already done for you. I can work with you to handle the merge

will make this code as simple as the indicative quote version where you just send the `firmQuotes` as `allQuotes` because they will already have the MakerUri attached. (Will also eliminate the need for `WrappedSignedNativeOrderMM`)

Reply:
Awesome, I've reverted changes related to RFQm and send them to https://github.com/0xProject/0x-rfq-api/pull/33.
It is much cleaner now